crates/rattler/src/install/clobber_registry.rs

```diff
@@ -0,0 +1,648 @@
+
+impl Default for ClobberRegistry {
+    fn default() -> Self {
+        Self {
+            paths_registry: HashMap::new(),
+            clobbers: HashMap::new(),
+            package_names: Vec::new(),
+            drop_bomb: DropBomb::new(
+                "did not call post_process on InstallDriver / ClobberRegistry",
+            ),
+        }
+    }
+}
+
+static CLOBBER_TEMPLATE: &str = "__clobber-from-";
+
+fn clobber_template(package_name: &PackageName) -> String {
+    format!("{CLOBBER_TEMPLATE}{}", package_name.as_normalized())
+}
+
+impl ClobberRegistry {
+    /// Create a new clobber registry that is initialized with the given prefix records.
+    pub fn from_prefix_records(prefix_records: &[PrefixRecord]) -> Self {
+        let mut registry = Self::default();
+
+        let mut temp_clobbers = Vec::new();
+        for prefix_record in prefix_records {
+            registry
+                .package_names
+                .push(prefix_record.repodata_record.package_record.name.clone());
+
+            let clobber_ending =
+                clobber_template(&prefix_record.repodata_record.package_record.name);
+
+            for p in &prefix_record.files {
+                if p.to_string_lossy().ends_with(&clobber_ending) {
+                    // register a clobbered path
+                    if let Some(full_file_name) = p.file_name() {
+                        if let Some((file_name, originating_package)) = full_file_name
+                            .to_string_lossy()
+                            .split_once(CLOBBER_TEMPLATE)
+                        {
+                            let path = p.with_file_name(file_name);
+                            temp_clobbers.push((path, originating_package.to_string()));
+                        }
+                    }
+                } else {
+                    registry
+                        .paths_registry
+                        .insert(p.clone(), registry.package_names.len() - 1);
+                }
+            }
+        }
+
+        for (path, originating_package) in temp_clobbers.iter() {
+            let idx = registry
+                .package_names
+                .iter()
+                .position(|n| n.as_normalized() == *originating_package)
+                .unwrap();
+
+            registry
+                .clobbers
+                .entry(path.clone())
+                .or_insert_with(|| vec![*registry.paths_registry.get(path).unwrap()])
+                .push(idx);
+        }
+
+        registry
+    }
+
+    fn clobber_name(path: &Path, package_name: &PackageName) -> PathBuf {
+        let file_name = path.file_name().unwrap_or_default();
+        let mut new_path = path.to_path_buf();
+        new_path.set_file_name(format!(
+            "{}{}",
+            file_name.to_string_lossy(),
+            clobber_template(package_name),
+        ));
+        new_path
+    }
+
+    /// Register the paths of a package when linking the package.
+    pub fn register_paths(
```

Comment:
Would be nice to describe a little more when this function is called (before actually linking files) and what the returned value is.